src/tests/test_raft_repl_dev.cpp

```diff
@@ -269,7 +269,12 @@ class TestReplicatedDB : public homestore::ReplDevListener {
                    "[Replica={}] Save logical snapshot callback obj_id={} term={} idx={} is_last={} num_items={}",
                    g_helper->replica_num(), snp_data->offset, s->get_last_log_term(), s->get_last_log_idx(),
                    snp_data->is_last_obj, num_items);
-        snp_data->offset = last_committed_idx + 1;
+
```

Comment:
we need to error out if `snp_data->is_last_obj == true` but `last_data_committed_lsn != s->get_last_log_idx()`, that means we have bug on sender side.

Reply:
actually, last_data_committed_lsn should always be smaller than get_last_log_idx(), since at least we have config log entry, which will be added to log_idx but will not be add to  last_data_committed_lsn